components/feature/script/src/main/java/mozilla/components/feature/script/SlowLoadingFeature.kt

```diff
@@ -0,0 +1,63 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package mozilla.components.feature.script
+
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.cancel
+import kotlinx.coroutines.flow.mapNotNull
+import mozilla.components.browser.state.selector.findTabOrCustomTabOrSelectedTab
+import mozilla.components.browser.state.store.BrowserStore
+import mozilla.components.concept.engine.script.SlowScriptCallback
+import mozilla.components.lib.state.ext.flowScoped
+import mozilla.components.support.base.feature.LifecycleAwareFeature
+import mozilla.components.support.base.log.logger.Logger
+import mozilla.components.support.ktx.kotlinx.coroutines.flow.ifChanged
+
+/**
+ * Feature implementation that provides a notification when the
+ * slowScript callback is called and enables the user to
+ * continue or stop running the script.
+ *
+ * @property sessionId id of the session running the slow script.
+ * @property store reference to the application's [BrowserStore].
+ *
+ * This is where we'd observe state.content.slowLoadingScriptCallback.
+ * When it's null -> everything is okay, nothing to do
+ * When is not null -> post a notification (what is that?) for user to choose next actions:
```

Comment:
> (what is that?)

On desktop, you can see this feature in action when an offending website has slow running javascript on it. On desktop, they show [this UI](https://pbs.twimg.com/media/DHvsYQ5XkAApVCR.jpg) bar right below the toolbar with actions the user can perform on the website. 

Similarly, the same mechanism is triggered from GeckoView and forwarded to us to prompt the user accordingly.

Hope that helps clarify it for you to figure out how write this doc. 🙂

Reply:
This was initially added as a `TODO` comment in expectation of clarifying what the notification should [look like](https://github.com/mozilla-mobile/fenix/issues/14959#issuecomment-1069172216). 
The screenshot you provided helped, thank you!